mypy/semanal.py

```diff
@@ -2983,12 +2993,16 @@ def infer_reachability_of_if_statement(s: IfStmt,
                                        platform: str) -> None:
     for i in range(len(s.expr)):
         result = infer_if_condition_value(s.expr[i], pyversion, platform)
-        if result == ALWAYS_FALSE:
+        if result in (ALWAYS_FALSE, MYPY_FALSE):
             # The condition is always false, so we skip the if/elif body.
             mark_block_unreachable(s.body[i])
```

Comment:
Even though PEP 484 only gives a positive example, it clearly states that TYPE_CHECKING

> is considered ``True`` during type checking (or other static analysis) but ``False`` at runtime.

I don't see any ambiguity there.